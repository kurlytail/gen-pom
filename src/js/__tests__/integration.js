import { execSync } from 'child_process';

describe('# integration test', () => {
    beforeEach(() => {
        execSync('rm -rf testoutput');
    });

    it('## should print help options', () => {
        const output = execSync('./scripts/sgen-pom.sh -h').toString();
        expect(output).toMatchSnapshot();
    });

    it('## should generate design', () => {
        const output = execSync('./scripts/sgen-pom.sh -d src/test/fixture/design.js -o testoutput').toString();
        expect(output).toMatchSnapshot();
    });

    it('## should generate design with merge', () => {
        let output = execSync(
            './scripts/sgen-pom.sh -d src/test/fixture/design.js -o testoutput --overwrite=merge'
        ).toString();
        expect(output).toMatchSnapshot();
        output = execSync(
            './scripts/sgen-pom.sh -d src/test/fixture/design.js -o testoutput --overwrite=merge'
        ).toString();
        expect(output).toMatchSnapshot();
    });

    it('## should generate design and run pom commands', () => {
        let output = execSync(
            './scripts/sgen-pom.sh -d src/test/fixture/design.js -o testoutput --overwrite=merge'
        ).toString();
        expect(output).toMatchSnapshot();
        output = execSync('/usr/local/bin/mvn package', { cwd: 'testoutput' }).toString();
    });
});
