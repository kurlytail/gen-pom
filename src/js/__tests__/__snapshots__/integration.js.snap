// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# integration test ## should generate design 1`] = `
"info: loading extension ./scripts/../templates/extensions/pom-group-id._
info: loading extension ./scripts/../templates/extensions/pom-artifact-id._
info: Generating testoutput/.settings/org.eclipse.jdt.core.prefs from template templates/.settings/org.eclipse.jdt.core.prefs._
info: Generating testoutput/.settings/org.eclipse.m2e.core.prefs from template templates/.settings/org.eclipse.m2e.core.prefs._
info: Generating testoutput/src/main/java/.gitkeep from template templates/src/main/java/.gitkeep._
info: Generating testoutput/src/main/resources/.gitkeep from template templates/src/main/resources/.gitkeep._
info: Generating testoutput/src/test/java/.gitkeep from template templates/src/test/java/.gitkeep._
info: Generating testoutput/src/test/resources/.gitkeep from template templates/src/test/resources/.gitkeep._
info: Generating testoutput/src/main/java/com/brainspeedtech/authentication/designs/Authentication.java from template templates/src/main/java/master-design-class.java._
info: Generating testoutput/src/main/java/com/brainspeedtech/authentication/flows/Register.java from template templates/src/main/java/master-flow-class.java._
info: Generating testoutput/src/main/java/com/brainspeedtech/authentication/tubes/DatabaseSinkResponder.java from template templates/src/main/java/master-tube-class.java._
info: Generating testoutput/src/main/java/com/brainspeedtech/authentication/tubes/ResponderGatewaySink.java from template templates/src/main/java/master-tube-class.java._
info: Generating testoutput/src/main/java/com/brainspeedtech/authentication/tubes/RecordDataDatabaseSink.java from template templates/src/main/java/master-tube-class.java._
info: Generating testoutput/src/main/java/com/brainspeedtech/authentication/tubes/RegisterPost.java from template templates/src/main/java/master-tube-class.java._
info: Generating testoutput/src/main/java/com/brainspeedtech/authentication/cells/GatewaySink.java from template templates/src/main/java/master-cell-class.java._
info: Generating testoutput/src/main/java/com/brainspeedtech/authentication/cells/DatabaseSink.java from template templates/src/main/java/master-cell-class.java._
info: Generating testoutput/src/main/java/com/brainspeedtech/authentication/cells/GatewaySource.java from template templates/src/main/java/master-cell-class.java._
info: Generating testoutput/src/main/java/com/brainspeedtech/authentication/cells/Responder.java from template templates/src/main/java/master-cell-class.java._
info: Generating testoutput/src/main/java/com/brainspeedtech/authentication/cells/RecordData.java from template templates/src/main/java/master-cell-class.java._
info: Generating testoutput/src/main/java/com/brainspeedtech/authentication/flows/Enum.java from template templates/src/main/java/master-flow-enum.java._
info: Generating testoutput/.classpath from template templates/.classpath._
info: Generating testoutput/.project from template templates/.project._
info: loading extension ./scripts/../templates/extensions/pom-model-version._
info: loading extension ./scripts/../templates/extensions/pom-version._
info: loading extension ./scripts/../templates/extensions/pom-name._
info: loading extension ./scripts/../templates/extensions/pom-description._
info: loading extension ./scripts/../templates/extensions/pom-property-reactor-core.version._
info: loading extension ./scripts/../templates/extensions/pom-property-junit-jupiter.version._
info: loading extension ./scripts/../templates/extensions/pom-property-java.version._
info: loading extension ./scripts/../templates/extensions/pom-dependencies-reactor-core.dependency._
info: loading extension ./scripts/../templates/extensions/pom-dependencies-junit-jupiter.dependency._
info: loading extension ./scripts/../templates/extensions/pom-dependencies-junit-jupiter-engine.dependency._
info: loading extension ./scripts/../templates/extensions/pom-dependencies-junit-jupiter-api.dependency._
info: Generating testoutput/pom.xml from template templates/pom.xml._
info: Generating testoutput/.gitignore from template templates/.gitignore._
info: Generating testoutput/Jenkinsfile from template templates/Jenkinsfile._
info: Generating testoutput/settings.xml from template templates/settings.xml._
info: Generating testoutput/maven.pem from template templates/maven.pem._
"
`;

exports[`# integration test ## should generate design and run pom commands 1`] = `
"info: loading extension ./scripts/../templates/extensions/pom-group-id._
info: loading extension ./scripts/../templates/extensions/pom-artifact-id._
info: Generating testoutput/.settings/org.eclipse.jdt.core.prefs from template templates/.settings/org.eclipse.jdt.core.prefs._
info: Generating testoutput/.settings/org.eclipse.m2e.core.prefs from template templates/.settings/org.eclipse.m2e.core.prefs._
info: Generating testoutput/src/main/java/.gitkeep from template templates/src/main/java/.gitkeep._
info: Generating testoutput/src/main/resources/.gitkeep from template templates/src/main/resources/.gitkeep._
info: Generating testoutput/src/test/java/.gitkeep from template templates/src/test/java/.gitkeep._
info: Generating testoutput/src/test/resources/.gitkeep from template templates/src/test/resources/.gitkeep._
info: Generating testoutput/src/main/java/com/brainspeedtech/authentication/designs/Authentication.java from template templates/src/main/java/master-design-class.java._
info: Generating testoutput/src/main/java/com/brainspeedtech/authentication/flows/Register.java from template templates/src/main/java/master-flow-class.java._
info: Generating testoutput/src/main/java/com/brainspeedtech/authentication/tubes/DatabaseSinkResponder.java from template templates/src/main/java/master-tube-class.java._
info: Generating testoutput/src/main/java/com/brainspeedtech/authentication/tubes/ResponderGatewaySink.java from template templates/src/main/java/master-tube-class.java._
info: Generating testoutput/src/main/java/com/brainspeedtech/authentication/tubes/RecordDataDatabaseSink.java from template templates/src/main/java/master-tube-class.java._
info: Generating testoutput/src/main/java/com/brainspeedtech/authentication/tubes/RegisterPost.java from template templates/src/main/java/master-tube-class.java._
info: Generating testoutput/src/main/java/com/brainspeedtech/authentication/cells/GatewaySink.java from template templates/src/main/java/master-cell-class.java._
info: Generating testoutput/src/main/java/com/brainspeedtech/authentication/cells/DatabaseSink.java from template templates/src/main/java/master-cell-class.java._
info: Generating testoutput/src/main/java/com/brainspeedtech/authentication/cells/GatewaySource.java from template templates/src/main/java/master-cell-class.java._
info: Generating testoutput/src/main/java/com/brainspeedtech/authentication/cells/Responder.java from template templates/src/main/java/master-cell-class.java._
info: Generating testoutput/src/main/java/com/brainspeedtech/authentication/cells/RecordData.java from template templates/src/main/java/master-cell-class.java._
info: Generating testoutput/src/main/java/com/brainspeedtech/authentication/flows/Enum.java from template templates/src/main/java/master-flow-enum.java._
info: Generating testoutput/.classpath from template templates/.classpath._
info: Generating testoutput/.project from template templates/.project._
info: loading extension ./scripts/../templates/extensions/pom-model-version._
info: loading extension ./scripts/../templates/extensions/pom-version._
info: loading extension ./scripts/../templates/extensions/pom-name._
info: loading extension ./scripts/../templates/extensions/pom-description._
info: loading extension ./scripts/../templates/extensions/pom-property-reactor-core.version._
info: loading extension ./scripts/../templates/extensions/pom-property-junit-jupiter.version._
info: loading extension ./scripts/../templates/extensions/pom-property-java.version._
info: loading extension ./scripts/../templates/extensions/pom-dependencies-reactor-core.dependency._
info: loading extension ./scripts/../templates/extensions/pom-dependencies-junit-jupiter.dependency._
info: loading extension ./scripts/../templates/extensions/pom-dependencies-junit-jupiter-engine.dependency._
info: loading extension ./scripts/../templates/extensions/pom-dependencies-junit-jupiter-api.dependency._
info: Generating testoutput/pom.xml from template templates/pom.xml._
info: Generating testoutput/.gitignore from template templates/.gitignore._
info: Generating testoutput/Jenkinsfile from template templates/Jenkinsfile._
info: Generating testoutput/settings.xml from template templates/settings.xml._
info: Generating testoutput/maven.pem from template templates/maven.pem._
"
`;

exports[`# integration test ## should generate design with merge 1`] = `
"info: loading extension ./scripts/../templates/extensions/pom-group-id._
info: loading extension ./scripts/../templates/extensions/pom-artifact-id._
info: Generating testoutput/.settings/org.eclipse.jdt.core.prefs from template templates/.settings/org.eclipse.jdt.core.prefs._
info: Generating testoutput/.settings/org.eclipse.m2e.core.prefs from template templates/.settings/org.eclipse.m2e.core.prefs._
info: Generating testoutput/src/main/java/.gitkeep from template templates/src/main/java/.gitkeep._
info: Generating testoutput/src/main/resources/.gitkeep from template templates/src/main/resources/.gitkeep._
info: Generating testoutput/src/test/java/.gitkeep from template templates/src/test/java/.gitkeep._
info: Generating testoutput/src/test/resources/.gitkeep from template templates/src/test/resources/.gitkeep._
info: Generating testoutput/src/main/java/com/brainspeedtech/authentication/designs/Authentication.java from template templates/src/main/java/master-design-class.java._
info: Generating testoutput/src/main/java/com/brainspeedtech/authentication/flows/Register.java from template templates/src/main/java/master-flow-class.java._
info: Generating testoutput/src/main/java/com/brainspeedtech/authentication/tubes/DatabaseSinkResponder.java from template templates/src/main/java/master-tube-class.java._
info: Generating testoutput/src/main/java/com/brainspeedtech/authentication/tubes/ResponderGatewaySink.java from template templates/src/main/java/master-tube-class.java._
info: Generating testoutput/src/main/java/com/brainspeedtech/authentication/tubes/RecordDataDatabaseSink.java from template templates/src/main/java/master-tube-class.java._
info: Generating testoutput/src/main/java/com/brainspeedtech/authentication/tubes/RegisterPost.java from template templates/src/main/java/master-tube-class.java._
info: Generating testoutput/src/main/java/com/brainspeedtech/authentication/cells/GatewaySink.java from template templates/src/main/java/master-cell-class.java._
info: Generating testoutput/src/main/java/com/brainspeedtech/authentication/cells/DatabaseSink.java from template templates/src/main/java/master-cell-class.java._
info: Generating testoutput/src/main/java/com/brainspeedtech/authentication/cells/GatewaySource.java from template templates/src/main/java/master-cell-class.java._
info: Generating testoutput/src/main/java/com/brainspeedtech/authentication/cells/Responder.java from template templates/src/main/java/master-cell-class.java._
info: Generating testoutput/src/main/java/com/brainspeedtech/authentication/cells/RecordData.java from template templates/src/main/java/master-cell-class.java._
info: Generating testoutput/src/main/java/com/brainspeedtech/authentication/flows/Enum.java from template templates/src/main/java/master-flow-enum.java._
info: Generating testoutput/.classpath from template templates/.classpath._
info: Generating testoutput/.project from template templates/.project._
info: loading extension ./scripts/../templates/extensions/pom-model-version._
info: loading extension ./scripts/../templates/extensions/pom-version._
info: loading extension ./scripts/../templates/extensions/pom-name._
info: loading extension ./scripts/../templates/extensions/pom-description._
info: loading extension ./scripts/../templates/extensions/pom-property-reactor-core.version._
info: loading extension ./scripts/../templates/extensions/pom-property-junit-jupiter.version._
info: loading extension ./scripts/../templates/extensions/pom-property-java.version._
info: loading extension ./scripts/../templates/extensions/pom-dependencies-reactor-core.dependency._
info: loading extension ./scripts/../templates/extensions/pom-dependencies-junit-jupiter.dependency._
info: loading extension ./scripts/../templates/extensions/pom-dependencies-junit-jupiter-engine.dependency._
info: loading extension ./scripts/../templates/extensions/pom-dependencies-junit-jupiter-api.dependency._
info: Generating testoutput/pom.xml from template templates/pom.xml._
info: Generating testoutput/.gitignore from template templates/.gitignore._
info: Generating testoutput/Jenkinsfile from template templates/Jenkinsfile._
info: Generating testoutput/settings.xml from template templates/settings.xml._
info: Generating testoutput/maven.pem from template templates/maven.pem._
"
`;

exports[`# integration test ## should generate design with merge 2`] = `
"info: loading extension ./scripts/../templates/extensions/pom-group-id._
info: loading extension ./scripts/../templates/extensions/pom-artifact-id._
info: Generating testoutput/.settings/org.eclipse.jdt.core.prefs from template templates/.settings/org.eclipse.jdt.core.prefs._
info: Generating testoutput/.settings/org.eclipse.m2e.core.prefs from template templates/.settings/org.eclipse.m2e.core.prefs._
info: Generating testoutput/src/main/java/.gitkeep from template templates/src/main/java/.gitkeep._
info: Generating testoutput/src/main/resources/.gitkeep from template templates/src/main/resources/.gitkeep._
info: Generating testoutput/src/test/java/.gitkeep from template templates/src/test/java/.gitkeep._
info: Generating testoutput/src/test/resources/.gitkeep from template templates/src/test/resources/.gitkeep._
info: Generating testoutput/src/main/java/com/brainspeedtech/authentication/designs/Authentication.java from template templates/src/main/java/master-design-class.java._
info: Generating testoutput/src/main/java/com/brainspeedtech/authentication/flows/Register.java from template templates/src/main/java/master-flow-class.java._
info: Generating testoutput/src/main/java/com/brainspeedtech/authentication/tubes/DatabaseSinkResponder.java from template templates/src/main/java/master-tube-class.java._
info: Generating testoutput/src/main/java/com/brainspeedtech/authentication/tubes/ResponderGatewaySink.java from template templates/src/main/java/master-tube-class.java._
info: Generating testoutput/src/main/java/com/brainspeedtech/authentication/tubes/RecordDataDatabaseSink.java from template templates/src/main/java/master-tube-class.java._
info: Generating testoutput/src/main/java/com/brainspeedtech/authentication/tubes/RegisterPost.java from template templates/src/main/java/master-tube-class.java._
info: Generating testoutput/src/main/java/com/brainspeedtech/authentication/cells/GatewaySink.java from template templates/src/main/java/master-cell-class.java._
info: Generating testoutput/src/main/java/com/brainspeedtech/authentication/cells/DatabaseSink.java from template templates/src/main/java/master-cell-class.java._
info: Generating testoutput/src/main/java/com/brainspeedtech/authentication/cells/GatewaySource.java from template templates/src/main/java/master-cell-class.java._
info: Generating testoutput/src/main/java/com/brainspeedtech/authentication/cells/Responder.java from template templates/src/main/java/master-cell-class.java._
info: Generating testoutput/src/main/java/com/brainspeedtech/authentication/cells/RecordData.java from template templates/src/main/java/master-cell-class.java._
info: Generating testoutput/src/main/java/com/brainspeedtech/authentication/flows/Enum.java from template templates/src/main/java/master-flow-enum.java._
info: Generating testoutput/.classpath from template templates/.classpath._
info: Generating testoutput/.project from template templates/.project._
info: loading extension ./scripts/../templates/extensions/pom-model-version._
info: loading extension ./scripts/../templates/extensions/pom-version._
info: loading extension ./scripts/../templates/extensions/pom-name._
info: loading extension ./scripts/../templates/extensions/pom-description._
info: loading extension ./scripts/../templates/extensions/pom-property-reactor-core.version._
info: loading extension ./scripts/../templates/extensions/pom-property-junit-jupiter.version._
info: loading extension ./scripts/../templates/extensions/pom-property-java.version._
info: loading extension ./scripts/../templates/extensions/pom-dependencies-reactor-core.dependency._
info: loading extension ./scripts/../templates/extensions/pom-dependencies-junit-jupiter.dependency._
info: loading extension ./scripts/../templates/extensions/pom-dependencies-junit-jupiter-engine.dependency._
info: loading extension ./scripts/../templates/extensions/pom-dependencies-junit-jupiter-api.dependency._
info: Generating testoutput/pom.xml from template templates/pom.xml._
info: Generating testoutput/.gitignore from template templates/.gitignore._
info: Generating testoutput/Jenkinsfile from template templates/Jenkinsfile._
info: Generating testoutput/settings.xml from template templates/settings.xml._
info: Generating testoutput/maven.pem from template templates/maven.pem._
"
`;

exports[`# integration test ## should print help options 1`] = `
"Usage:
  node sgen [OPTION]

Options:
  -m, --map=ARG+       mapping files
  -d, --design=ARG+    design files
  -e, --extension=ARG+ extension directories or files
  -o, --output=ARG     output directory for generated files
      --overwrite=ARG  Overwrite modes, can be one of skip, regen, error or merge                         (default: skip)
      --forceOverwrite Override overwrite flags in the generator map with the one provided on --overwrite
  -h, --help           display this help
"
`;
